_docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/DownloadManagerApi.php

```diff
@@ -58,22 +58,34 @@ public function __construct(ClientInterface $client, EntityTypeManager $entity_t
    *   sucessful. Otherwise, returns FALSE.
    */
   public function getContentById($id) {
-    try {
-      $feed_url = $this->apiUrl . $id;
-      $request = $this->client->request('GET', $feed_url);
-      $response = $request->getBody()->getContents();
+    $cid = 'download_manager_api:product:' . $id;
 
-      return json_decode($response);
+    // Serve the cached Download Manager data if available.
+    if ($cache = \Drupal::cache()->get($cid)) {
+      $data = $cache->data;
     }
-    catch (\Exception $e) {
-      \Drupal::logger('rhd_assemblies')->error(
-        "Exception while calling Download Manager API: @message", [
-          '@message' => $e->getMessage(),
-        ]
-      );
+    // If this is uncached or the cache is invalid, fetch fresh data.
+    else {
+      try {
+        $feed_url = $this->apiUrl . $id;
+        $request = $this->client->request('GET', $feed_url);
+        $response = $request->getBody()->getContents();
+        $data = json_decode($response);
+        // Persist this data to the download_manager_api cache bin for an hour.
+        \Drupal::cache()->set($cid, $data, time() + 3600);
```

Comment:
@jordanpagewhite Could we extract this into a configurable value e.g. `$config['redhat_developers']['cache.ttl']['download_manager'] = <time_in_seconds>` with some sane default if ^ is not set? 

In terms of the actual TTL for this data, can you remind me what this integration is actually used for? Is it simply to get product names? If _yes_ that I'm gonna say we can go for a longer TTL on this one.